src/routes/clinical-proxy.ts

```diff
@@ -0,0 +1,48 @@
+import urlJoin from 'url-join';
+import { CLINICAL_SERVICE_ROOT, SUBMISSION_TEMPLATE_PATH } from '../config';
+import logger from '../utils/logger';
+import express from 'express';
+import { Request, Response } from 'express';
+const router = express.Router();
+
+import { createProxyMiddleware } from 'http-proxy-middleware';
+
+// Our specification download service can't use GraphQL because GraphQL specification requires the content-type
+// that it returns be json, and we want to be able to return other content types, such as tab-separated-values,
+// so that the user is automatically prompted to save the file from their browser.
+
+// 'all' will retrieve the zip file with all templates
+// for specific templates 'templateName'.tsv or 'templateName' will get the tsv from clinical
+const templatePathRewrite = (pathName: string, req: Request) => {
+  const name = req.params.template.replace(/.tsv$/, '');
+  return urlJoin(SUBMISSION_TEMPLATE_PATH, name);
+};
+router.use(
+  '/template/:template',
+  createProxyMiddleware({
+    target: CLINICAL_SERVICE_ROOT,
+    pathRewrite: templatePathRewrite,
+    onError: handleError,
+  }),
+);
+
+const programTsvExportPathRewrite = (pathName: string, req: Request) => {
+  const programId = req.params.programId;
+  return urlJoin('/clinical/program/', programId, '/tsv-export');
+};
+router.use(
+  '/program/:programId/all-clinical-data',
+  createProxyMiddleware({
+    target: CLINICAL_SERVICE_ROOT,
+    pathRewrite: programTsvExportPathRewrite,
+    onError: handleError,
+  }),
+);
+
+// This is for handling nodejs/system errors (e.g. connection failed)
+function handleError(err: Error, req: Request, res: Response) {
+  logger.error('Clinical Router Error - ' + err);
+  return res.status(500).send('Internal Server Error');
+}
```

Comment:
lol this guy is working thanks to javascript's magical function hoisting. I personally prefer the `const handleError = ...` declaration and having this declared up top because it's a bit more explicit, but that might just be me... Just wanna point out that's what's happening for consideration

Reply:
I'm good with that :+1:. Will move it to the top.